tests/console/oprofile.pm

```diff
@@ -0,0 +1,42 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Simple tests for oprofile.
+# - profile example application (fio to create some load)
+# - check report
+# Maintainer: Michael Grifalconi <mgrifalconi@suse.com>
+
+use strict;
+use warnings;
+use base 'consoletest';
+use testapi;
+use utils;
+
+sub run {
+    select_console 'root-console';
+
+    # Install fio as load generator and oprofile
+    zypper_call "in fio oprofile";
+
+    # Start a system wide profiling
+    type_string "operf --system-wide &";
+
+    # Start load and stop profiler when finished
+    assert_script_run "fio --name=randwrite --filename=/tmp/randwrite --bs=4k --size=1G && killall -s SIGINT operf";
+
+    # Make sure the report is generated correctly and also check for an entry for the fio load
+    assert_script_run "opreport";
+    assert_script_run "opreport | grep 'fio'";
+
+    # Cleanup
```

Comment:
Is in this test case, not `operf --system-wide &` still running in the background afterwards? Perhaps add 

    script_run('kill %1');

either here or to `post_run_hook` and `post_fail_hook` so the background job gets terminated (if this is not happening in `opreport`)

Reply:
operf will be stopped right after generating the load: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10656/files#diff-834b5e2564819e5c1c6602b98f60bc5fR31

Now that you point that out, I guess it makes more sense to change `&&` with `;` to make sure the kill command runs even if `fio` fails.

Reply:
If the question was about the behavior of `SIGINT`, according to what I tried, operf will gracefully stop and exit:
```
root@d505:/var/lib/openqa/tests/os-autoinst-distri-opensuse(oprofile)# operf --system-wide &                                           
[1] 16894

root@d505:/var/lib/openqa/tests/os-autoinst-distri-opensuse(oprofile)# operf: Press Ctl-c or 'kill -SIGINT 16894' to stop profiling    
operf: Profiler started

root@d505:/var/lib/openqa/tests/os-autoinst-distri-opensuse(oprofile)#                                                                 
root@d505:/var/lib/openqa/tests/os-autoinst-distri-opensuse(oprofile)# killall -s SIGINT operf                                         
root@d505:/var/lib/openqa/tests/os-autoinst-distri-opensuse(oprofile)# * * * * WARNING: Profiling rate was throttled back by the kernel
* * * *
The number of samples actually recorded is less than expected, but is                                                                  
probably still statistically valid.  Decreasing the sampling rate is the                                                               
best option if you want to avoid throttling.

WARNING: Lost samples detected! See /var/lib/openqa/share/tests/os-autoinst-distri-opensuse/oprofile_data/samples/operf.log for details.
Lowering the sampling rate may reduce or eliminate lost samples.
See the '--events' option description in the operf man page for help.                                                                  

Profiling done.

[1]+  Done                    operf --system-wide
```